app/src/crypto.c

```diff
@@ -154,7 +145,8 @@ uint16_t digest_message(uint8_t *digest, uint16_t digestMax, const uint8_t *mess
             cx_sha3_init(&sha3, 256);
             cx_hash((cx_hash_t*)&sha3, CX_LAST, message, messageLen, digest, 32);
             zemu_log_stack("sha3_256 ready");
-            return 32;
+            *digest_size = 32;
```

Comment:
Another clean up: it would be good to use a macro instead of `32` in all the sha3 branching, just like `CX_SHA256_SIZE` in the sha2 case.